client/my-sites/plugins/plugin-remove-button/index.jsx

```diff
@@ -57,9 +60,74 @@ module.exports = React.createClass( {
 		}
 	},
 
-	renderButton: function() {
-		var inProgress = PluginsLog.isInProgressAction( this.props.site.ID, this.props.plugin.slug, [ 'REMOVE_PLUGIN' ] );
+	getDisabledInfo: function() {
+		if ( ! this.props.site ) { // we don't have enough info
+			return null;
+		}
+
+		if ( ! this.props.site.hasMinimumJetpackVersion ) {
+			return this.translate( '%(site)s is not running an up to date version of Jetpack', {
+				args: { site: this.props.site.title }
+			} );
+		}
+
+		if ( this.props.site.options.is_multi_network ) {
+			return this.translate( '%(site)s is part of a multi-network installation, which is not currently supported.', {
+				args: { site: this.props.site.title }
+			} );
+		}
+
+		if ( this.props.site.options.unmapped_url !== this.props.site.options.main_network_site ) {
+			return this.translate( 'Only the main site on a multi-site installation can remove plugins.', {
+				args: { site: this.props.site.title }
+			} );
+		}
+
+		if ( ! this.props.site.canUpdateFiles && this.props.site.options.file_mod_disabled ) {
+			let reasons = utils.getSiteFileModDisableReason( this.props.site );
+			let html = [];
 
+			if ( reasons.length > 1 ) {
+				html.push(
+					<p key="reason-shell">
+						{ this.translate( 'Removing plugins is not available for %(site)s:', { args: { site: this.props.site.title } } ) }
+					</p>
+				);
+				let list = reasons.map( ( reason, i ) => ( <li key={ 'reason-i' + i + '-' + this.props.site.ID } >{ reason }</li> ) );
+				html.push( <ul className="plugin-action__disabled-info-list" key="reason-shell-list">{ list }</ul> );
+			} else {
+				html.push(
+					<p key="reason-shell">{
+						this.translate( 'Removing plugins is not available for %(site)s. %(reason)s', {
+							args: { site: this.props.site.title, reason: reasons[0] }
+						} )
+					}</p> );
+			}
+			html.push(
+				<ExternalLink
+					key="external-link"
+					onClick={
+						analytics.ga.recordEvent.bind( this, 'Plugins', 'Clicked How do I fix disabled autoupdates' )
+					}
+					href="https://jetpack.me/support/site-management/#file-update-disabled"
+					>
+					{ this.translate( 'How do I fix this?' ) }
+				</ExternalLink>
+			);
+
+			return html;
+		}
+		return null;
+	},
+
+	renderButton: function() {
+		var inProgress = PluginsLog.isInProgressAction( this.props.site.ID, this.props.plugin.slug, [ 'REMOVE_PLUGIN' ] ),
+			getDisabledInfo = this.getDisabledInfo(),
+			label = getDisabledInfo
+				? this.translate( 'Removeing disabled', {
```

Comment:
`removing` instead of `removeing`? Not sure if 'removeing' is correct or not, though, so don't mind me if so :)